### PR TITLE
Fix Wrong Intrinsic gas cost in `eth_sendRawTransaction` precheck calculation

### LINKED ISSUE #1631: Wrong Intrinsic Gas Cost in `eth_sendRawTransaction` precheck
### Description

Wrong intrinsic gas is calculated by `eth_sendRawTransaction` precheck in all cases. For example deploying [this](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/tools/hardhat-example/contracts/Greeter.sol) greeter contract takes around 79-80k gas, but relay calculates that it needs at least 160k. This results in error throw in the relay even before the transaction reaches consensus node.

This bug didn't surface until now, because we've used hardcoded values for estimateGas, now that we have precise calculation of estimate gas from the mirror-node it appears that relay calculations were wrong.

### Steps to reproduce

1. Deploy contract using image from [this](https://github.com/hashgraph/hedera-mirror-node/pull/6633) PR, which now calculates correct estimateGas

### Additional context

_No response_

### Hedera network

other

### Version

main

### Operating system

None

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:
Wrong intrinsic gas is calculated by eth_sendRawTransaction precheck in all cases. For example deploying [this](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/tools/hardhat-example/contracts/Greeter.sol) greeter contract takes around 79-80k gas, but relay calculates that it needs at least 160k. This results in error throw in the relay even before the transaction reaches consensus node.

This bug didn't surface until now, because we've used hardcoded values for estimateGas, now that we have precise calculation of estimate gas from the mirror-node it appears that relay calculations were wrong.
Calculation done according to Hedera [docs](https://docs.hedera.com/hedera/core-concepts/smart-contracts/gas-and-fees#gas-fees)

Removed estimateGas workaround for the mirror-node bug, as it's fixed there.

⚠️ Should merge and release this after https://github.com/hashgraph/hedera-mirror-node/pull/6633 is ready and deployed on public network.
**Related issue(s)**:

Fixes #1631 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/precheck.ts [hunk 1]
```diff
@@ -186,26 +186,24 @@ export class Precheck {
...
+            zeros++;
+        } else {
+            nonZeros++;
+        }
     }
 
-    const nonZeros = data.replace('0x', '').length - zeros;
-    const cost = constants.TX_BASE_COST + constants.TX_DATA_ZERO_COST * zeros + constants.ISTANBUL_TX_DATA_NON_ZERO_COST * nonZeros;
-    return isCreate ? cost + constants.TX_CREATE_EXTRA : cost;
+    return constants.TX_BASE_COST + constants.TX_DATA_ZERO_COST * zeros + constants.ISTANBUL_TX_DATA_NON_ZERO_COST * nonZeros;
```
Comment (AlfredoG87) [inline packages/relay/src/lib/precheck.ts:207]: One question, what is the `IntrinsicGasCost`? and why is it calculated with this formula? 
I thought that accurate GasCost was possible only using contracts/call?

Comment (georgi-l95) [reply packages/relay/src/lib/precheck.ts:207]: This is like the minimum amiunt of gas needed to execute the transaction, based on the data provided and base cost for every transaction. The real and more accurate gas estimation is done by the web3 module. Here we are just making sure that the provided gasLimit is covering the bare minimum for this tx. Calucalation is done based on the info from Hedera docs, but basicaly for every empty byte we multiply by 4, and for every non empty we multiply by 16 and add all this to the base cost (21000).

Comment (AlfredoG87) [reply packages/relay/src/lib/precheck.ts:207]: Thanks for the explanation, it makes sense 👍

**Comment by Nana-EC** [comment]
@georgi-l95 0.87.0 us in mainent now so you should be able to rebase

File: packages/relay/tests/lib/precheck.spec.ts [hunk 2]
```diff
@@ -427,4 +427,37 @@ describe('Precheck', async function () {
...
+    it('should be able to able to calculate transfer', function () {
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(transfer);
+      expect(intrinsicGasCost).to.be.equal(constants.TX_BASE_COST);
+    });
+
+    it('should be able to calculate for odd length tx', function () {
+      const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(invalidTx);
+      expect(intrinsicGasCost).to.be.greaterThan(constants.TX_BASE_COST);
+    });
+  });
```
Comment (Nana-EC) [inline packages/relay/tests/lib/precheck.spec.ts:462]: Can you add some more fine tuned examples that confirm exact numbers for a a small and medium set of contract bytes e.g.
- empty contract (smallest permissable)
- simple contract with a singleslot that is set on constructor
- you can also use existing test contracts in the repo

Doesn't have to be exhaustive but 1 or 2 to know the number calculation work for the intrinsic gas calculation on a create and not on create scenario

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice, some suggestions

File: packages/relay/src/lib/precheck.ts [hunk 3]
```diff
@@ -223,29 +223,26 @@ export class Precheck {
...
       }
     }
 
-    const nonZeros = data.replace('0x', '').length - zeros;
-    const cost =
-      constants.TX_BASE_COST +
-      constants.TX_DATA_ZERO_COST * zeros +
-      constants.ISTANBUL_TX_DATA_NON_ZERO_COST * nonZeros;
-    return isCreate ? cost + constants.TX_CREATE_EXTRA : cost;
+    return (
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply packages/relay/src/lib/precheck.ts:244]: Synced offline. Adding here would make us inconsistent with the mirror node

File: packages/relay/tests/lib/precheck.spec.ts (hunk 2 above)
Comment (Nana-EC) [reply packages/relay/tests/lib/precheck.spec.ts:462]: @georgi-l95 can address this in follow up

**Review by Nana-EC — APPROVED** [review]
LG
Minor comments and tests requested. Please address in follow up PR

**Pull request merged**